src/quo2/components/wallet/account_card/view.cljs

```diff
@@ -45,72 +47,90 @@
                                      :margin-top
                                      10)}])]))
 
-(defn- user-account
-  [{:keys [state name balance percentage-value loading? amount customization-color type emoji metrics?
-           theme on-press]}]
-  (let [watch-only?        (= :watch-only type)
-        empty-type?        (= :empty type)
-        account-amount     (if (= :empty state) "€0.00" amount)
-        account-name       (if (= :empty state) (i18n/label :t/Account 1) name)
-        account-percentage (if (= :empty state) "€0.00" percentage-value)]
-    (if loading?
-      [loading-view
-       {:customization-color customization-color
-        :type                type
-        :theme               theme
-        :metrics?            metrics?}]
-      [rn/pressable
-       {:style    (style/card customization-color watch-only? metrics? theme)
-        :on-press on-press}
-       [rn/view {:style style/profile-container}
-        [rn/view {:style {:padding-bottom 2 :margin-right 2}}
-         [text/text {:style style/emoji} emoji]]
-        [rn/view {:style style/watch-only-container}
-         [text/text
-          {:size   :paragraph-2
-           :weight :medium
-           :style  (style/account-name watch-only? theme)}
-          account-name]
-         (when watch-only? [icon/icon :reveal {:color colors/neutral-50 :size 12}])]]
-       [text/text
-        {:size   :heading-2
-         :weight :semi-bold
-         :style  (style/account-value watch-only? theme)}
-        balance]
-       (when metrics?
-         [rn/view {:style style/metrics-container}
-          [text/text
-           {:weight              :semi-bold
-            :size                :paragraph-2
-            :accessibility-label :metrics
-            :style               (style/metrics watch-only? theme)}
-           account-percentage]
-          (when (not empty-type?)
-            [:<>
-             [rn/view (style/separator watch-only? theme)]
+(def user-account
```

Comment:
Why is this a def now? 🤔

@ulisesmac wdyt?

Reply:
This is not a proper form-2 component, we should be using `defn`.

With `def`, the state is initialized once for the lifetime of the anonymous function. This doesn't allow the component to be used twice or more on the same screen because they'd be writing/reading the same state in memory.

With `defn` and the form-2 style, the state would be initialized on every mount. Quite different.

Reply:
Done

Reply:
Yes exactly as @ilmotta said, the state atom is always the same no matter if we try to mount/remount the component.

BTW, I think the component generated would be a `form-1` one, IMO very interesting, because the only way to notice this is not correct is when when we remount the component and its state is preserved